test/src/test/java/org/corfudb/integration/LogReplicationReaderWriterIT.java

```diff
@@ -354,9 +355,41 @@ public static void writeLogEntryMsgs(List<LogReplicationEntryMsg> msgQ, Set<Stri
             log.debug("msgQ is EMPTY");
         }
 
-        for (LogReplicationEntryMsg msg : msgQ) {
+        Iterator<LogReplicationEntryMsg> msgQIterator = msgQ.iterator();
+        LogReplicationEntryMsg msg = null;
+        //Scenario 1: sink process the msg normally
+        if (msgQIterator.hasNext()) {
+            msg = msgQIterator.next();
             writer.apply(msg);
         }
+
+        // Scenario 2: Sink processes the previous msg, but Source is unware due to an interruption which causes LR to stop and start
+        // thereby resetting the Source's metadata to reflect the last ACK.Which means, the metadata on Source and Sink may not match.
+        // The source now resends the same data + new data that was written in the meanwhile, and sink filters and process only the new data.
+
+        if (msgQIterator.hasNext()) {
+            LogReplicationEntryMsg oldMsg = msg;
+            msg = msgQIterator.next();
+            LogReplication.LogReplicationEntryMetadataMsg newMetadata = LogReplication.LogReplicationEntryMetadataMsg.newBuilder()
```

Comment:
is there anything like clone method we can use in protoBuf?

Reply:
Changed to using "mergeFrom"